sync/src/chain.rs

```diff
@@ -881,8 +891,17 @@ impl ChainSync {
 			self.on_peer_aborting(sync, peer_id);
 		}
 	}
+
 	/// Called when peer sends us new transactions
-	fn on_peer_transactions(&mut self, _io: &mut SyncIo, _peer_id: PeerId, _r: &UntrustedRlp) -> Result<(), PacketDecodeError> {
+	fn on_peer_transactions(&mut self, io: &mut SyncIo, peer_id: PeerId, r: &UntrustedRlp) -> Result<(), PacketDecodeError> {
+		let chain = io.chain();
+		let item_count = r.item_count();
+		trace!(target: "sync", "{} -> Transactions ({} entries)", peer_id, item_count);
+		let fetch_latest_nonce = |a : &Address| chain.nonce(a);
```

Comment:
When syncing from start it panics when asking for `nonce` with
```
TrieDB::new(d6d2…d52c): Trie root not found!
```
in: `state::State::from_existing`

I will be investigating what happens there.